.github/scripts/version_updater.rb

```diff
@@ -0,0 +1,123 @@
+
+                    puts "Updating #{gem_name} with: #{bundler_args}"
+                    run_bundler_command(gemfile, bundler_args)
+                else
+                    puts "The latest (#{latest}) of #{gem_name} is NOT satisfied by #{gem_requirements}"
+                    # TODO - should we attempt to do a --conservative update here?
+                end
+            end
+        end
+    end
+
+
+    def parse_paths(paths)
+        paths.each do |path|
+            parser = Bundler::LockfileParser.new(Bundler.read_file(path))
+            parser.specs.each do |spec|
+                if @versions[spec.name]
+                    # "Gemfile.lock" -> "Gemfile"
+                    gem_file = path.sub(/\.lock$/, '')
+                    # gems_in['grpc'].Add([Gemfile, 1.0.0])
+                    @gems_in[spec.name] << [gem_file, spec.version]
+                    # keep track of the highest vesion seen - TODO likley redunant now and replaced by "gems_in"
+                    @versions[spec.name] = [@versions[spec.name], spec.version].max
+                else # The gem doesn't appear to be an integration - likely a dependency of one though - skip it
+                    # puts "Skipping #{spec.name} as it isn't in contrib"
+                end
+            end
+        end
+    end
+
+    def main
+        # for each directory in /contrib/ read the hash of the loaded Gems
+        # https://www.rubydoc.info/github/rubygems/rubygems/Gem.loaded_specs
+        # taken from: https://github.com/DataDog/apm-shared-github-actions/blob/2b49e71feba54bdfaabdeeb026ec3032d819371f/.github/scripts/get-tested-integration-versions.rb#L36
+        ddtrace_specs = `grep -Rho 'Gem.loaded_specs.*' lib/datadog/tracing/contrib/`
+        integrated_gems = ddtrace_specs.split.map { |m| m.match(/Gem.loaded_specs\[.([^\]]+).\]/)&.[](1) }.uniq.compact
+
+        # populate each found integration with a dummy version of "0.0.0"
+        integrated_gems.each do |integrated|
+            @versions[integrated] = Gem::Version.new('0.0.0')
+        end
+
+        integrated_gems.each do |name, version|
+            puts "#{name}: #{version}"
+        end
+
+        paths = Dir['gemfiles/jruby_*.lock', 'gemfiles/ruby_*.lock']
+        
+        # TODO - a lock file can contain the same gem multiple times
+        # for example: grpc (1.66.0), grpc (1.66.0-aarch64-linux), and grpc (1.66.0-x86_64-linux)
+        # are all "grpc", but each one is "different", so the same gemfile in this loop gets added multiple times
+        # should probably fix this
+        parse_paths(paths)
+        # TODO - could we look through open PR titles to rule out which gems to update?
+        #        if the PR is the autogenerated PR to update the gem we could skip it
+        gem_to_update = integrated_gems.sample # HACKY pick a random gem to attempt to update
```

Comment:
Picking a random gem seems fine, maybe we want to be able to query github to see which PRs are already open and skip if a gem has already been added / updated